contracts/contracts/strategies/ConvexStrategy.sol

```diff
@@ -78,37 +79,27 @@ contract ConvexStrategy is BaseCurveStrategy {
         require(success, "Failed to deposit to Convex");
     }
 
-    function _lpWithdraw(uint256 numPTokens) internal override {
+    function _lpWithdraw(uint256 numCrvTokens) internal override {
+        uint256 gaugePTokens = IRewardStaking(cvxRewardStakerAddress).balanceOf(
+            address(this)
+        );
+
+        // Not enough in this contract or in the Gauge, can't proceed
```

Comment:
Why do we do this additional check? If we don't have enough tokens, the withdraw will revert regardless?

Reply:
This is just to get a nicer error message instead of the "reverted without a reason" message. You think it is better to optimise for gas here?